pageserver/src/tenant/timeline.rs

```diff
@@ -1709,10 +1709,7 @@ impl Timeline {
         guard.initialize_local_layers(loaded_layers, disk_consistent_lsn + 1);
 
         if let Some(rtc) = self.remote_client.as_ref() {
-            rtc.schedule_layer_file_deletion(&needs_cleanup)?;
-            rtc.schedule_index_upload_for_file_changes()?;
-            // Tenant::create_timeline will wait for these uploads to happen before returning, or
-            // on retry.
+            rtc.flushing_delete_layers(&needs_cleanup).await?;
```

Comment:
New thread for performance implications, which is why I am not jumping into merging this: this would or could slow down startup. Basically never have we depended on an actual remote storage action at startup and we've always scheduled instead.

I wonder if we should just:
- increment a counter on these `schedule_layer_file_deletion`
- rename `schedule_layer_file_deletion` as `schedule_future_layer_deletion` to make it clear it's to be used for only one thing
- before starting the first uploads, wait for deletion queue to have flushed if we deleted anything
    - alternatively this could be a barrier scheduled after uploading the index_part
    - I do not remember if deletionqueue returned such but it probably could

Reply:
+1 to not blocking startup on the network

Reply:
This is in the per-tenant task though.
What are you fearing exactly?